starter-code/src/functions-and-arrays.js

```diff
@@ -10,16 +14,39 @@ var words = [
   'orchard',
   'crackpot'
 ];
+function findLongestWord(myArray){
+  if(myArray.length == 0) return undefined;
+  if(myArray.length == 1) return myArray[0];
+  if(myArray.length == 2) return myArray[0];
+  return myArray.reduce((acc,currVal)=>{
+  return (acc.length > currVal.length) ? acc : currVal},0
+  )
+  }
 
 // Calculating a Sum
-
+function sumArray(myArray){
+  if(myArray.length == 0) return 0;
+  if(myArray.length == 1) return myArray[0];
+  if(myArray.every(el=>el===0)) return 0;
+  return myArray.reduce((acc,currVal) => acc + currVal,0);
+}
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
 // Calculate the Average
-
+function averageNumbers(myArray) {
+  if(myArray.length == 0) return undefined;
```

Comment:
again here, you're adding unneeded code: 

```
function sumArray(myArray) {
  if (myArray.length === 0) return 0;

  return myArray.reduce((acc, currVal) => acc + currVal, 0);
} 
```

While not needed, it's better to use `===` rather than `==`